src/Discord/Parts/WebSockets/MessageReaction.php

```diff
@@ -211,4 +211,47 @@ protected function getGuildAttribute(): ?Guild
 
         return null;
     }
+    
+    /**
+     * Delete this reaction
+     *
+     * @param int|null $type The type of deletion to perform.
+     *
+     * @return ExtendedPromiseInterface
+     */
+    public function delete(?int $type = null): ExtendedPromiseInterface
+    {
+        if (is_null($type)) {
+            if ($this->user_id == $this->discord->id) {
+                $type = Message::REACT_DELETE_ME;
+            } else {
+                $type = Message::REACT_DELETE_ID;
+            }
+        }
+
+        $types = [Message::REACT_DELETE_ALL, Message::REACT_DELETE_ME, Message::REACT_DELETE_ID, Message::REACT_DELETE_EMOJI];
+
+        $emoticon = $this->emoji->toReactionString();
+
+        if (in_array($type, $types)) {
+            switch ($type) {
+                case Message::REACT_DELETE_ALL:
+                    $url = Endpoint::bind(Endpoint::MESSAGE_REACTION_ALL, $this->channel_id, $this->message_id);
+                    break;
+                case Message::REACT_DELETE_ME:
+                    $url = Endpoint::bind(Endpoint::OWN_MESSAGE_REACTION, $this->channel_id, $this->message_id, $emoticon);
+                    break;
+                case Message::REACT_DELETE_ID:
+                    $url = Endpoint::bind(Endpoint::USER_MESSAGE_REACTION, $this->channel_id, $this->message_id, $emoticon, $this->user_id);
+                    break;
+                case Message::REACT_DELETE_EMOJI:
+                    $url = Endpoint::bind(Endpoint::MESSAGE_REACTION_EMOJI, $this->channel_id, $this->message_id, $emoticon);
+                    break;
+            }
+
+            return $this->http->delete($url);
+        }
+
+        return \React\Promise\reject();
```

Comment:
Could we possibly reject with an exception? Or just throw an exception.

Reply:
Actually, I don't know why :D I was just copying the code from [Message::deleteReaction()](https://github.com/discord-php/DiscordPHP/blob/master/src/Discord/Parts/Channel/Message.php#L652).

Reply:
Ill just remove the type checking, since this part is made for one message reaction by user, any invalid type will just default to `REACTION_DELETE_ID`